app/Helpers/database/site-award.php

```diff
@@ -243,3 +244,44 @@ function getUserEventAwardCount(string $user): int
 
     return $dataOut['TotalAwards'];
 }
+
+/**
+ * Retrieves a target user's site award metadata for a given game ID.
+ * An array is returned with keys "beaten-softcore", "beaten-hardcore",
+ * "completed", and "mastered", which contain corresponding award details.
+ * If no progression awards are found, or if the target username is not provided,
+ * no awards are fetched or returned.
+ *
+ * @return array the array of a target user's site award metadata for a given game ID
+ */
+function getUserGameProgressionAwards(int $gameId, string $username): array
```

Comment:
Extracted from the `GameCard` component class so this functionality can be reused elsewhere.